python/cudf/cudf/core/series.py

```diff
@@ -2402,10 +2404,30 @@ def _concat(cls, objs, axis=0, index=True):
                     )
 
             if dtype_mismatch:
-                if isinstance(objs[0]._column, cudf.core.column.DecimalColumn):
+                if all(
+                    [
+                        isinstance(obj._column, cudf.core.column.DecimalColumn)
+                        for obj in objs
+                    ]
+                ):
                     objs = _decimal_normalize_types(*objs)
-                else:
+                elif all([is_numerical_dtype(obj.dtype) for obj in objs]):
                     objs = numeric_normalize_types(*objs)
+                elif all(
+                    [
+                        isinstance(obj._column, cudf.core.column.DecimalColumn)
+                        or is_numerical_dtype(obj.dtype)
+                        for obj in objs
+                    ]
+                ):
+                    decimal_type = _find_common_type_decimal(
+                        [
+                            obj.dtype
+                            for obj in objs
+                            if is_decimal_dtype(obj.dtype)
+                        ]
+                    )
+                    objs = [obj.astype(decimal_type) for obj in objs]
```

Comment:
Is it possible for all this logic to be handled in `find_common_type`? For example, it'd be nice to be able to replace this with just:

```python
common_dtype = find_common_type(objs)
objs = [obj.astype(common_type) for obj in objs
```